src/assets/TokenListController.ts

```diff
@@ -224,87 +199,46 @@ export class TokenListController extends BaseController<
     }, this.intervalDelay);
   }
 
-  /**
-   * Fetching token list.
-   */
-  async fetchTokenList(): Promise<void> {
-    if (this.useStaticTokenList) {
-      await this.fetchFromStaticTokenList();
-    } else {
-      await this.fetchFromDynamicTokenList();
-    }
-  }
-
-  /**
-   * Fetching token list from the contract-metadata as a fallback.
-   */
-  async fetchFromStaticTokenList(): Promise<void> {
-    const tokenList: TokenListMap = {};
-    for (const tokenAddress in contractMap) {
-      const {
-        erc20,
-        logo: filePath,
-        ...token
-      } = (contractMap as ContractMap)[tokenAddress];
-      if (erc20) {
-        tokenList[tokenAddress] = {
-          ...token,
-          address: tokenAddress,
-          iconUrl: filePath,
-          occurrences: null,
-        };
-      }
-    }
-
-    this.update(() => {
-      return {
-        tokenList,
-        tokensChainsCache: {},
-      };
-    });
-  }
-
   /**
    * Fetching token list from the Token Service API.
    */
-  async fetchFromDynamicTokenList(): Promise<void> {
+  async fetchTokenList(): Promise<void> {
     const releaseLock = await this.mutex.acquire();
     try {
-      const cachedTokens: TokenListToken[] | null = await safelyExecute(() =>
+      const { tokensChainsCache } = this.state;
+      let tokenList: TokenListMap = {};
+      const cachedTokens: TokenListMap = await safelyExecute(() =>
         this.fetchFromCache(),
       );
-      const { tokensChainsCache, ...tokensData } = this.state;
-      const tokenList: TokenListMap = {};
       if (cachedTokens) {
-        for (const token of cachedTokens) {
-          tokenList[token.address] = token;
-        }
+        // Use non-expired cached tokens
+        tokenList = cachedTokens;
       } else {
-        const tokensFromAPI: DynamicToken[] = await safelyExecute(() =>
+        // Fetch fresh token list
+        const tokensFromAPI: RawToken[] = await safelyExecute(() =>
           fetchTokenList(this.chainId, this.abortController.signal),
         );
+
         if (!tokensFromAPI) {
-          const backupTokenList = tokensChainsCache[this.chainId]
-            ? tokensChainsCache[this.chainId].data
-            : [];
-          for (const token of backupTokenList) {
-            tokenList[token.address] = token;
-          }
+          // Fallback to expired cached tokens
+          tokenList = tokensChainsCache?.[this.chainId]?.data || {};
```

Comment:
What do you think about setting a default value for `tokensChainsCache` so that we don't have to check whether it's there or not?

Reply:
We're currently setting a default value on the `tokenList` level but not within the `.data` level. Not sure if it makes sense to set a default on the `data` level since that would indicate that `tokenList` was previously populated. Did I read that right?

Reply:
Makes sense that you wouldn't be able to provide a default value on the `.data` level. I meant on a higher level. My thought is, would it be possible to drop the first `?.` and say `tokenList = tokensChainsCache[this.chainId]?.data || {};`?

Reply:
Ah I see. I realized that we are already setting the default on the `tokensChainsCache` so I just removed optional chaining immediately after whenever we're using it.